### ISSUE TITLE
repro?

### ISSUE THREAD

**Issue by tachim** [bbrattoli/JigsawPuzzlePytorch#1]
Hi @bbrattoli -- just checking, were you actually able to get this to train? I'm trying to overfit on a small subset of ImageNet (see below) but the loss becomes nan unfortunately.

```ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00028932.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00006172.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00047236.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00033634.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00042206.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00022814.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00048145.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00017111.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00006257.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00017333.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00036231.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00022808.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00047438.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00047250.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00023027.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00004731.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00013475.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00038938.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00031336.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00025652.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00041514.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00014257.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00014589.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00035579.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00049435.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00024587.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00001408.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00036235.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00003069.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00044678.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00009586.JPEG
ImageNet/ILSVRC2012_img_val/val/ILSVRC2012_val_00006834.JPEG
```

```
JigsawPuzzlePytorch$ python JigsawTrain.py --classes=1000 --lr=0.1 --hamming=mean --batch 10 ~/ImageNet/
CPU mode
Start training: lr 0.100000, batch size 10, classes 1000
Checkpoint: checkpoints/
TESTING: 0), Accuracy 0.00%
Learning Rate 0.100000
[1/70] 0) LR 0.10000, Loss: 6.903, Accuracy 0.0%
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
TESTING: 25), Accuracy 0.00%
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
TESTING: 50), Accuracy 0.00%
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
TESTING: 75), Accuracy 0.00%
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
Learning Rate 0.100000
Saved: checkpoints/
TESTING: 100), Accuracy 0.00%
Learning Rate 0.010000
[21/70] 100) LR 0.01000, Loss: nan, Accuracy 0.0%
Saved: checkpoints/
Learning Rate 0.010000
Saved: checkpoints/
Learning Rate 0.010000
Saved: checkpoints/
Learning Rate 0.010000
Saved: checkpoints/
Learning Rate 0.010000
Saved: checkpoints/
TESTING: 125), Accuracy 0.00%
Learning Rate 0.010000
```

**Comment by tachim** [comment]
Hmm, nevermind, it seems that using `gpu` training fixed this. Strange.
